# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.h
# Commit: c5112a80ad2f
# Full Hash: c5112a80ad2f910f06889c9cadecc8c4728cd858
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-11-08 21:36:02
# Description:
#   Bug 1799495 - Ensure WaitForShmem has a valid DrawTargetWebgl to access. r=aosmond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161554
# ==============================================================================

diff -r 879b0989791c -r c5112a80ad2f dom/canvas/DrawTargetWebgl.h
--- a/dom/canvas/DrawTargetWebgl.h	Tue Nov 08 16:21:48 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.h	Tue Nov 08 16:23:56 2022 +0000
@@ -309,7 +309,7 @@
     void ClearEmptyTextureMemory();
     void ClearCachesIfNecessary();
 
-    void WaitForShmem();
+    void WaitForShmem(DrawTargetWebgl* aTarget);
   };
 
   RefPtr<SharedContext> mSharedContext;
@@ -486,7 +486,7 @@
 
   void WaitForShmem() {
     if (mSharedContext->mWaitForShmem) {
-      mSharedContext->WaitForShmem();
+      mSharedContext->WaitForShmem(this);
     }
   }
 